src/main/java/seedu/momentum/model/ProjectBookWithUi.java

```diff
@@ -51,4 +51,14 @@ public boolean isTagsVisible() {
         return isTagsVisible;
     }
 
+    @Override
+    public boolean equals(Object other) {
+        return super.equals(other) && (other == this // short circuit if same object
+                || (other instanceof ProjectBookWithUi // instanceof handles nulls
+                && viewMode.equals(((ProjectBookWithUi) other).getViewMode())
+                && ((project == null && ((ProjectBookWithUi) other).getProject() == null)
+                || (project.equals(((ProjectBookWithUi) other).getProject())))
+                && isTagsVisible == ((ProjectBookWithUi) other).isTagsVisible()));
+        //&& predicate.equals(((ProjectBookWithUi) other).getPredicate())));
```

Comment:
Is this commented out line needed

Reply:
I need to clarify with the others on whether we need this line, because this line conflicts with tests when it shouldn't for certain tests. I commented it out first